web/docs/general/typescript.md

```diff
@@ -0,0 +1,151 @@
+---
+title: TypeScript Support
+---
+
+import TypescriptServerNote from '../_TypescriptServerNote.md'
+
+# TypeScript support
+
+TypeScript is a programming language that brings static type analysis to JavaScript. It is a superset of JavaScript (i.e., all valid JavaScript programs are valid TypeScript programs) and compiles to JavaScript before running.
+
+TypeScript's type system detects common errors at build time (reducing the chance of runtime errors in production) and enables type-based auto-completion in IDEs.
+
+Each Wasp feature comes with accompanying TypeScript documentation.
+
+If you're starting a new project and want to use TypeScript, you don't need to do anything special.
+Just follow the feature docs you are interested in, and they will tell you everything you need to know.
+We recommend you start by going through [the tutorial](../tutorial/01-create.md).
+
+If you already have a Wasp project written in JavaScript and you wish to
+migrate it to TypeScript, then this guide is meant for you.
```

Comment:
Some language simplification:
```suggestion
TypeScript is a programming language that adds static type analysis to JavaScript. It is a superset of JavaScript, meaning all JavaScript code is valid TypeScript code, and it compiles to JavaScript.

TypeScript's type system helps catch errors at build time, reducing runtime errors, and provides type-based auto-completion in IDEs.

Each Wasp feature includes TypeScript documentation.

If you're starting a new project and want to use TypeScript, you don't need to do anything special.
Just follow the feature docs you are interested in, and they will tell you everything you need to know.
We recommend you start by going through [the tutorial](../tutorial/01-create.md).

To migrate an existing Wasp project from JavaScript to TypeScript, follow this guide.
```

I'd definitely go for more basic English since I suspect it's easier to read through (especially non-native speakers).